roles/wordpress/tasks/main.yml

```diff
@@ -1,60 +1,88 @@
 ---
-- name: Download WordPress
-  get_url: url=http://wordpress.org/wordpress-{{ wp_version }}.tar.gz dest=/srv/wordpress-{{ wp_version }}.tar.gz
-           sha256sum="{{ wp_sha256sum }}"
-
-- name: Extract archive
-  command: chdir=/srv/ /bin/tar xvf wordpress-{{ wp_version }}.tar.gz creates=/srv/wordpress
-
-- name: Add group "wordpress"
-  group: name=wordpress
-
-- name: Add user "wordpress"
-  user: name=wordpress group=wordpress home=/srv/wordpress/
-
-- name: Fetch random salts for WordPress config
-  local_action: command curl https://api.wordpress.org/secret-key/1.1/salt/
-  register: "wp_salt"
-  become: no
-
-- name: Create WordPress database
+- name: install dependencies and SELinux tools
+  package:
+    state: present
+    name:
+      - MySQL-python
+      - libselinux-python
+      - policycoreutils-python
+
+- name: create web server directory
+  file:
+    state: directory
+    path: /var/www
+
+- name: download and extract archive
+  unarchive:
+    creates: /var/www/wordpress/
+    dest: /var/www/
+    remote_src: yes
+    src: https://wordpress.org/latest.tar.gz
+
+- name: add group "wordpress"
+  group:
+    name: wordpress
+
+- name: add user "wordpress"
+  user:
+    name: wordpress
+    group: wordpress
+    home: /var/www/wordpress/
+
+- name: register random salts for WordPress config
+  register: wp_salt
+  command: curl https://api.wordpress.org/secret-key/1.1/salt/
+  # If there is a clean way to do this with either get_url or uri, please
+  # switch to the module for a Ansible-native way of doing this.
+  # uri:
+  #   return_content: yes
+  #   url: https://api.wordpress.org/secret-key/1.1/salt/
```

Comment:
This looks correct... `uri` with `register: wp_salt`, `wp_salt['content']` should be the thing you want.

If you really want a var with only the content for some reason, you can:
```
set_fact:
    wp_salt: "{{ wp_salt.content }}"
```